packages/react-instantsearch/src/core/createConnector.js

```diff
@@ -105,7 +105,7 @@ export default function createConnector(connectorDesc) {
         props: this.getProps(nextProps),
       });
 
-      if (isWidget) {
+      if (isWidget && !isEqual(nextProps, this.props)) {
```

Comment:
So we are fixing a bug somehow here? Do you think we can add a test that checks the number of times search is called? I would rather try to fix that in another PR also because this one can be focused only on InstantSearch multi guide + removing the constraint